tests/test_config_parsing/test_ecl_config.py

```diff
@@ -17,4 +17,59 @@ def test_ecl_config_dict_creates_equal_config(config_dict):
     to_config_file(filename, config_dict)
     res_config = ResConfig(user_config_file=filename)
     config_dict[ConfigKeys.CONFIG_DIRECTORY] = cwd
-    assert res_config.ecl_config == EclConfig(config_dict=config_dict)
+    assert res_config.ecl_config == EclConfig.from_dict(config_dict=config_dict)
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_ecl_config_fails_on_non_sensical_refcase_file():
+    refcase_file = "CEST_PAS_UNE_REFCASE"
+    refcase_file_content = """
+_________________________________________     _____    ____________________
+\\______   \\_   _____/\\_   _____/\\_   ___ \\   /  _  \\  /   _____/\\_   _____/
+ |       _/|    __)_  |    __)  /    \\  \\/  /  /_\\  \\ \\_____  \\  |    __)_
+ |    |   \\|        \\ |     \\   \\     \\____/    |    \\/        \\ |        \\
+ |____|_  /_______  / \\___  /    \\______  /\\____|__  /_______  //_______  /
+        \\/        \\/      \\/            \\/         \\/        \\/         \\/
+"""
+    with open(refcase_file, "w+", encoding="utf-8") as refcase_file_handler:
+        refcase_file_handler.write(refcase_file_content)
+    try:
+        EclConfig(refcase_file=refcase_file)
```

Comment:
Also here we can use
```python
with pytest.raises(ValueError, match=refcase_file):
  _ = EclConfig(refase_file=refcase_file)
```

match can also contain other parts that should be in the error message.

Reply:
sure thing!

Reply:
done!